SupportClient/SupportClient/Views/FeedbackView.swift

```diff
@@ -0,0 +1,295 @@
+            typeDropdown.centerYAnchor.constraint(equalTo: typeLabel.centerYAnchor),
+            typeDropdown.heightAnchor.constraint(equalToConstant: 32)
+        ])
+        NSLayoutConstraint.activate([
+            messageLabel.leadingAnchor.constraint(equalTo: leadingAnchor, constant: padding),
+            messageLabel.topAnchor.constraint(equalTo: typeLabel.bottomAnchor, constant: 2 * padding)
+        ])
+        NSLayoutConstraint.activate([
+            messageTextView.topAnchor.constraint(equalTo: messageLabel.bottomAnchor, constant: padding),
+            messageTextView.heightAnchor.constraint(equalToConstant: 100),
+            messageTextView.leadingAnchor.constraint(equalTo: leadingAnchor, constant: padding),
+            messageTextView.trailingAnchor.constraint(equalTo: trailingAnchor, constant: -padding)
+        ])
+        NSLayoutConstraint.activate([
+            attachmentsLabel.leadingAnchor.constraint(equalTo: leadingAnchor, constant: padding),
+            attachmentsLabel.topAnchor.constraint(equalTo: messageTextView.bottomAnchor, constant: 2 * padding)
+        ])
+        NSLayoutConstraint.activate([
+            addFileButton.leadingAnchor.constraint(equalTo: leadingAnchor, constant: padding),
+            addFileButton.topAnchor.constraint(equalTo: attachmentsLabel.bottomAnchor, constant: padding),
+            addFileButton.widthAnchor.constraint(equalToConstant: 100),
+            addFileButton.heightAnchor.constraint(equalToConstant: imageHeight)
+        ])
+        NSLayoutConstraint.activate([
+            attachmentsCollectionView.heightAnchor.constraint(equalTo: addFileButton.heightAnchor),
+            attachmentsCollectionView.leadingAnchor.constraint(equalTo: addFileButton.trailingAnchor, constant: padding),
+            attachmentsCollectionView.trailingAnchor.constraint(equalTo: trailingAnchor, constant: -padding),
+            attachmentsCollectionView.topAnchor.constraint(equalTo: addFileButton.topAnchor)
+        ])
+    }
+    
+    func checkStatus() {
+        if let _ = typeDropdown.selectedIndex {
+            if messageTextView.textColor != UIColor.lightGray || attachedFiles.count > 0 {
+                status = .complete
+            } else {
+                status = .incomplete
+            }
+        }
+    }
+    
+    // MARK: - OBJC Functions
+    @objc func dismissKeyboard (_ sender: UITapGestureRecognizer) {
+        self.endEditing(true)
+    }
+    
+    @objc func addFile(_ sender: UIButton) {
+        let imagePicker = ImagePickerController()
+        imagePicker.settings.theme.selectionStyle = .checked
+        imagePicker.settings.fetch.assets.supportedMediaTypes = [.image, .video]
+        currentViewController()!.presentImagePicker(imagePicker,
+            select: nil,
+            deselect: nil,
+            cancel: nil,
+            finish: { assets in
+                self.attachedFiles = Array(Set(self.attachedFiles + assets.map { $0 }))
```

Comment:
Y do we need the `assets.map { $0 }` call here

Reply:
This takes all the PHAssets selected, combines it w/ the ones already selected, turns it into a set of PHAssets to remove duplicates, then back into an array

Reply:
Can't we just do `Array(Set(self.attachedFiles + assets))`? It seems to me like the in-place mapping is unnecessary unless I'm missing something